setup.js

```diff
@@ -0,0 +1,108 @@
+import { Nation } from "./constants";
+
+const error = (want) => (x) => {
+  throw new Error(`got=${x.value}, want=${want}`);
+};
+
+export default ({ players }) => {
+  const nationAssignments = {
+    3: ({ id, nation }) =>
+      nation.when({
+        AH: () => [
+          { id, nation: Nation.AH },
+          { id, nation: Nation.GB },
+        ],
+        IT: () => [
+          { id, nation: Nation.IT },
+          { id, nation: Nation.RU },
+        ],
+        FR: () => [
+          { id, nation: Nation.FR },
+          { id, nation: Nation.GE },
+        ],
+        GB: error("AH|IT|FR"),
+        RU: error("AH|IT|FR"),
+        GE: error("AH|IT|FR"),
+      }),
+    2: ({ id, nation }) =>
+      nation.when({
+        AH: () => [
+          { id, nation: Nation.AH },
+          { id, nation: Nation.FR },
+          { id, nation: Nation.GE },
+        ],
+        IT: () => [
+          { id, nation: Nation.IT },
+          { id, nation: Nation.RU },
+          { id, nation: Nation.GB },
+        ],
+        FR: error("AH|IT"),
+        GB: error("AH|IT"),
+        RU: error("AH|IT"),
+        GE: error("AH|IT"),
+      }),
+    4: (x) => [x],
+    5: (x) => [x],
+    6: (x) => [x],
+  };
+
+  const that = {
+    players: {},
+    nations: new Map(
+      Array.from(Nation).map((n) => [
+        n,
+        { treasury: 0, controller: null, rondelPosition: null },
+      ])
+    ),
+    order: players.map((p) => p.id),
+  };
+
+  players
+    .map(nationAssignments[players.length])
+    .flat()
```

Comment:
Wait, `flat()` works?? I thought this was the stuff of legends...

Reply:
We're [living in the future](https://caniuse.com/#feat=array-flat), baby